src/main/java/generator/ModelBuilder.java

```diff
@@ -17,9 +17,9 @@
     private RandomPositioner randomPositioner;
     private MyFactory factory;
 
-    public ModelBuilder(Model model) throws FileNotFoundException, OWLOntologyCreationException {
+    public ModelBuilder(Model model, MyFactory factory) throws FileNotFoundException, OWLOntologyCreationException {
         this.model = model;
-        this.factory = MyFactorySingleton.getFactory();
+        this.factory = factory;
```

Comment:
Zmieniliśmy na przekazywanie factory w parametrze a nie w przez singleton

Reply:
Jak był singleton to byłoby trudne generowanie albo wczytywanie scenariusza z innego pliku niż ten domyślny

Reply:
Wgl moze zmieńmy MyFactory na jakieś OWLFactory albo coś takiego :)